### PR TITLE
Feat/nav

### PR DESCRIPTION
**Pull request by salunkemanas**
Screenshot 2024-10-13 at 10 13 47 PM
Added fixed side and top navbar

> Top bar height is too big
> Nav items are not visually aligned
> Search icon is too small

### CODE REVIEW

File: src/styles/components/ui/SideNav/SideNav.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import Box from "@/styles/components/ui/Box";
+import Stack from "@/styles/components/ui/Stack";
+import { Home, LineChart, Package, ShoppingCart, Users } from "lucide-react";
+
+export default function SideNav() {
+  const navElements = [
```
Comment (hatim-s) [inline src/styles/components/ui/SideNav/SideNav.tsx:8]: move into useMemo

File: src/styles/components/ui/SideNav/SideNav.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      className="bg-white h-full items-start shadow-md justify-center w-full"
+    >
+      {navElements.map((elem) => {
+        const Icon = elem.icon;
+        return (
+          <Box
+            key={elem.name}
+            className="flex items-center gap-x-2 w-full pl-4 py-2 text-gray-600 font-medium transition-all hover:text-blue-600 hover:bg-blue-50 hover:cursor-pointer"
+          >
+            <Icon className="size-5" />
```
Comment (hatim-s) [inline src/styles/components/ui/SideNav/SideNav.tsx]: @hatim-s todo: use semantic size

File: src/styles/components/ui/SideNav/SideNav.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            {elem.name}
```
Comment (hatim-s) [inline src/styles/components/ui/SideNav/SideNav.tsx]: use Typography

File: src/styles/components/ui/SideNav/SideNav.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <Stack
+      direction="column"
+      gapY="md"
+      className="bg-white h-full items-start shadow-md justify-center w-full"
+    >
+      {navElements.map((elem) => {
+        const Icon = elem.icon;
+        return (
+          <Box
```
Comment (hatim-s) [inline src/styles/components/ui/SideNav/SideNav.tsx]: use stack

File: src/styles/components/ui/Stack/Stack.tsx [hunk 5]
```diff
@@ -74,13 +74,18 @@ type StackProps = {
...
-    gapY ? stackClasses({ gapY }) : null
-  );
-  const mergedClasses = twMerge(baseClass, gapClasses, className);
+  const baseClass = stackClasses({
+    direction,
+    gap: gap || undefined,
+    gapX: gapX || undefined,
+    gapY: gapY || undefined,
+  });
+  // const gapClasses = cn(
```
Comment (hatim-s) [inline src/styles/components/ui/Stack/Stack.tsx:83]: remove

File: src/styles/components/ui/SideNav/SideNav.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    { name: "Dashboard", icon: Home },
+    { name: "Orders", icon: ShoppingCart },
+    { name: "Products", icon: Package },
+    { name: "Customers", icon: Users },
+    { name: "Analytics", icon: LineChart },
+  ];
+  return (
+    <Stack
+      direction="column"
+      gapY="md"
```
Comment (hatim-s) [inline src/styles/components/ui/SideNav/SideNav.tsx]: @hatim-s todo: remove gap from props and add semantic tokens

File: src/styles/components/ui/Stack/Stack.tsx [hunk 7]
```diff
@@ -74,13 +74,18 @@ type StackProps = {
...
+  //   gap ? stackClasses({ gap }) : null,
+  //   gapX ? stackClasses({ gapX }) : null,
+  //   gapY ? stackClasses({ gapY }) : null
+  // );
+  const mergedClasses = twMerge(baseClass, className);
```
Comment (hatim-s) [inline src/styles/components/ui/Stack/Stack.tsx:88]: @hatim-s  todo: check if clsx should be used?

File: src/styles/components/ui/TopNav/TopNav.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { Avatar, AvatarFallback, AvatarImage } from "../../base/avatar";
+import { Input } from "../../base/input";
+
+import Box from "../Box";
+import Stack from "../Stack";
+import { MagnifyingGlassIcon } from "@radix-ui/react-icons";
```
Comment (hatim-s) [inline src/styles/components/ui/TopNav/TopNav.tsx:8]: organize imports properly

File: src/styles/components/ui/TopNav/TopNav.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+export default function TopNav() {
+  return (
+    <Stack className="justify-between p-3 items-center px-6 bg-white w-screen">
+      <Box className="font-bold text-lg">Logo</Box>
```
Comment (hatim-s) [inline src/styles/components/ui/TopNav/TopNav.tsx]: use typography, and use actual logo

File: src/styles/components/ui/TopNav/TopNav.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+      <Stack className="w-1/3 rounded-md px-4 py-2 focus:outline-none focus:ring-2 focus:ring-blue-300 items-center">
+        <Input placeholder="Search" className="" />
+        <MagnifyingGlassIcon className="ml-2" />
```
Comment (hatim-s) [inline src/styles/components/ui/TopNav/TopNav.tsx]: see if margin can be avoided

File: src/styles/components/ui/TopNav/TopNav.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+      </Stack>
+      <Avatar className="w-10 h-10 rounded-full hover:shadow-md transition-shadow">
+        <AvatarImage src="https://github.com/shadcn.png" />
+        <AvatarFallback>CN</AvatarFallback>
```
Comment (hatim-s) [inline src/styles/components/ui/TopNav/TopNav.tsx]: use typograpphy

File: src/app/layout.tsx [hunk 12]
```diff
@@ -26,7 +30,21 @@ export default function RootLayout({
 }>) {
   return (
     <html lang="en">
-      <body className={`${geistSans.variable} ${geistMono.variable} antialiased`}>{children}</body>
+      <body
+        className={`${geistSans.variable} ${geistMono.variable} antialiased`}
+      >
+        <Stack direction="column" className="w-screen h-screen">
+          <Box className="h-20 w-full fixed top-0 left-0 z-50 bg-white border-b">
```
Comment (hatim-s) [inline src/app/layout.tsx]: @hatim-s todo add themes

File: src/app/layout.tsx [hunk 13]
```diff
@@ -26,7 +29,23 @@ export default function RootLayout({
...
     <html lang="en">
-      <body className={`${geistSans.variable} ${geistMono.variable} antialiased`}>{children}</body>
+      <body
+        className={`${geistSans.variable} ${geistMono.variable} antialiased`}
+      >
+        <Stack direction="column" className="w-screen h-screen overflow-hidden">
+          <Stack className="h-[65px] w-full bg-white border-b">
+            <TopNav />
+          </Stack>
+          <Stack className="size-full">
```
Comment (anandabhi04) [inline src/app/layout.tsx:39]: why h-full?

File: src/app/layout.tsx [hunk 14]
```diff
@@ -26,7 +29,23 @@ export default function RootLayout({
...
+            <Stack className="w-1/6 h-[calc(100vh-65px)] border-r justify-center">
+              <SideNav />
+            </Stack>
+            <Stack className="w-5/6 p-3 overflow-auto h-[calc(100vh-65px)]">
```
Comment (anandabhi04) [inline src/app/layout.tsx]: overflow-y

File: src/styles/components/ui/SideNav/SideNav.tsx [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+      className="bg-white size-full items-start shadow-md justify-center"
```
Comment (anandabhi04) [inline src/styles/components/ui/SideNav/SideNav.tsx]: size-full?

File: src/styles/components/ui/Stack/Stack.tsx [hunk 16]
```diff
@@ -74,13 +74,18 @@ type StackProps = {
...
-  const baseClass = stackClasses({ direction });
-  const gapClasses = cn(
-    gap ? stackClasses({ gap }) : null,
-    gapX ? stackClasses({ gapX }) : null,
-    gapY ? stackClasses({ gapY }) : null
-  );
-  const mergedClasses = twMerge(baseClass, gapClasses, className);
+  const baseClass = stackClasses({
+    direction,
+    gap: gap || undefined,
```
Comment (anandabhi04) [inline src/styles/components/ui/Stack/Stack.tsx:79]: undefined?

**Review by anandabhi04 — CHANGES_REQUESTED** [review]

File: src/app/layout.tsx (hunk 12 above)
Comment (hatim-s) [reply src/app/layout.tsx]: this is done partially with #10

**Review by hatim-s — APPROVED** [review]

**Pull request merged**